kmk/hid.py

```diff
@@ -55,13 +58,60 @@ class HIDUsagePage:
 
 
 class AbstractHID:
-    REPORT_BYTES = 8
+    report_bytes_default = 8
+    REPORT_BYTES = report_bytes_default
+    hid_devices = {}
 
     def __init__(self, **kwargs):
+        self._nkro = False
+        self._mouse = True
+        self._pan = False
+        self.find_devices()
+
+        self._cc_report = bytearray(HID_REPORT_SIZES[HIDReportTypes.CONSUMER] + 1)
+        self._cc_report[0] = HIDReportTypes.CONSUMER
+        self._cc_pending = False
+
+        self.setup_keyboard_hid()
+        self.setup_mouse_hid()
+
+        self.start_watchdog()
 
+    def show_debug(self):
+        if debug.enabled:
+            if self._nkro:
+                debug('use NKRO')
+            else:
+                debug('use 6KRO')
+            if self._mouse and self._pan:
+                debug('enable horizontal scrolling mouse')
+            elif self._mouse:
+                debug('enable mouse')
+            else:
+                debug('disable mouse')
+
+    def find_devices(self):
+        self.devices = {}
```

Comment:
This belongs in `__init__()`

Reply:
First we blank out the dict, then we repopulate it.